Libraries/xcsdk/Tools/xcrun.cpp

```diff
@@ -298,17 +298,70 @@ main(int argc, char **argv)
     /*
      * Determine the SDK to use.
      */
-    xcsdk::SDK::Target::shared_ptr target = manager->findTarget(SDK.value_or("macosx"));
-    if (target == nullptr) {
-        if (SDK) {
-            fprintf(stderr, "error: unable to find sdk '%s'\n", SDK->c_str());
+    const std::string defaultSDK = "macosx";
+    xcsdk::SDK::Target::shared_ptr target = manager->findTarget(SDK.value_or(defaultSDK));
+    if (verbose) {
+        if (target == nullptr) {
+            fprintf(stderr, "verbose: not using any SDK\n");
         } else {
-            fprintf(stderr, "error: unable to find default sdk\n");
+            fprintf(stderr, "verbose: using sdk '%s': %s\n", target->canonicalName().c_str(), target->path().c_str());
         }
-        return -1;
     }
-    if (verbose) {
-        fprintf(stderr, "verbose: using sdk '%s': %s\n", target->canonicalName().c_str(), target->path().c_str());
+
+    bool showSDKValue = options.showSDKPath() ||
+        options.showSDKVersion() ||
+        options.showSDKBuildVersion() ||
+        options.showSDKPlatformPath() ||
+        options.showSDKPlatformVersion();
+
+    /*
+     * Perform SDK-specific actions.
+     */
+    if (showSDKValue) {
+        if (target == nullptr) {
+            if (SDK) {
+                fprintf(stderr, "error: unable to find sdk '%s'\n", SDK->c_str());
+            } else {
+                fprintf(stderr, "error: unable to find default sdk '%s'\n", defaultSDK.c_str());
+            }
+            return -1;
+        }
+
+        if (options.showSDKPath()) {
+            printf("%s\n", target->path().c_str());
+        } else if (options.showSDKVersion()) {
+            printf("%s\n", target->version().c_str());
+        } else if (options.showSDKBuildVersion()) {
+            if (auto product = target->product()) {
+                printf("%s\n", product->buildVersion().c_str());
+            } else {
+                fprintf(stderr, "error: sdk has no build version\n");
+                return -1;
+            }
+        } else if (options.showSDKPlatformPath()) {
+            if (auto platform = target->platform()) {
+                printf("%s\n", platform->path().c_str());
+            } else {
+                fprintf(stderr, "error: sdk has no platform\n");
+                return -1;
+            }
+        } else if (options.showSDKPlatformVersion()) {
+            if (auto platform = target->platform()) {
+                printf("%s\n", platform->version().c_str());
+            } else {
+                fprintf(stderr, "error: sdk has no platform\n");
+                return -1;
+            }
+        }
+
+        return 0;
```

Comment:
The early return here makes the flow a little non-obvious to me, maybe wrap the code below in an `else` block?